src/FlatbufferMessage.h

```diff
@@ -9,56 +9,91 @@
 
 #pragma once
 
+#include "Msg.h"
 #include "logger.h"
 
 namespace FileWriter {
-class BufferTooSmallError : public std::runtime_error {
+class FlatbufferError : public std::runtime_error {
+public:
+  explicit FlatbufferError(std::string const &What)
+      : std::runtime_error(What) {}
+};
+
+class BufferTooSmallError : public FlatbufferError {
 public:
   explicit BufferTooSmallError(const std::string &what)
-      : std::runtime_error(what){};
+      : FlatbufferError(what){};
 };
 
-class UnknownFlatbufferID : public std::runtime_error {
+class UnknownFlatbufferID : public FlatbufferError {
 public:
   explicit UnknownFlatbufferID(const std::string &what)
-      : std::runtime_error(what){};
+      : FlatbufferError(what){};
 };
 
-class InvalidFlatbufferTimestamp : public std::runtime_error {
+class InvalidFlatbufferTimestamp : public FlatbufferError {
 public:
   explicit InvalidFlatbufferTimestamp(const std::string &what)
-      : std::runtime_error(what){};
+      : FlatbufferError(what){};
 };
 
-class NotValidFlatbuffer : public std::runtime_error {
+class NotValidFlatbuffer : public FlatbufferError {
 public:
   explicit NotValidFlatbuffer(const std::string &what)
-      : std::runtime_error(what){};
+      : FlatbufferError(what){};
 };
 
 /// \brief A wrapper around a databuffer which holds a flatbuffer.
 ///
 /// Used to simplify passing around flatbuffers and the most important pieces of
 /// meta-data.
-///
-/// \note Does not take ownership of any pointers. You must make sure to
-/// free the pointers you passed yourself.
 class FlatbufferMessage {
 public:
   using SrcHash = size_t;
 
   /// Constructor is used in unit testing code to simplify set-up.
   FlatbufferMessage() = default;
 
-  /// \brief Verifies the data in the flatbuffer to make sure if it is valid.
+  /// \brief Creates a flatbuffer message, verifies the message and extracts
+  /// metadata.
+  ///
+  /// \note This constructor should be used as little as possible and then only
+  /// in unit tests.
   ///
-  /// \param BufferPtr Pointer to memory location containing flatbuffer.
-  /// \param Size Size of flatbuffer in bytes.
+  /// \param BufferPtr Pointer to memory containing the data.
+  /// \param Size Number of bytes in message.
+  /// \note Will make a copy of the data in the Kafka message.
   FlatbufferMessage(uint8_t const *BufferPtr, size_t Size);
 
+  /// \brief Creates a flatbuffer message, verifies the message and extracts
+  /// metadata.
+  ///
+  /// \param KafkaMessage The Kafka message used to create the Flatbuffer
+  /// message. \note Will make a copy of the data in the Kafka message.
+  explicit FlatbufferMessage(FileWriter::Msg const &KafkaMessage);
+
+  /// \brief Creates a flatbuffer message, verifies the message and extracts
+  /// metadata.
+  ///
+  /// \param KafkaMessage The copy constructor.
```

Comment:
This doxygen comment seems to be completely wrong.

Reply:
Oopsie, fixes.